xwiki-platform-core/xwiki-platform-panels/xwiki-platform-panels-ui/src/main/resources/Panels/MyRecentModifications.xml

```diff
@@ -173,15 +173,33 @@
     <property>
       <content>{{velocity}}
 #set ($void = $services.async.useEntity("wiki:$xcontext.database"))
-#set ($query = $services.query.xwql('where doc.author = :author order by doc.date desc'))
+#set ($query = $services.query.hql('select doc.fullName, doc.language from XWikiDocument doc where doc.author = :author order by doc.date desc'))
 #set ($recentDocs = $query.addFilter('hidden').addFilter('unique').bindValue('author', $xcontext.user).setLimit(5).execute())
+## We count the occurence of a document to specify the document language when there are duplicates
+#set ($docNamesCounter = {})
+#foreach ($doc in $recentDocs)
+  #if ($docNamesCounter.containsKey($doc[0]))
+    #set ($docNamesCounter[$doc[0]] = $docNamesCounter[$doc[0]] + 1)
+  #else
+    #set ($docNamesCounter[$doc[0]] = 1)
+  #end
+#end
 #if ($recentDocs.size() &gt; 0 || $showEmptyPanels)
   #panelheader($services.localization.render('xe.panels.modifications.my'))
-  #foreach ($docName in $recentDocs)
-    #set ($recentDoc = $xwiki.getDocument($docName).getTranslatedDocument())
+  #foreach ($doc in $recentDocs)
+    #set ($recentDoc = $xwiki.getDocument($doc[0]).getTranslatedDocument($doc[1]))
     #if ($recentDoc)
+      #set ($docLocale = $recentDoc.locale)
+      #if ("$docLocale" == '')
+        #set ($docLocale = $recentDoc.defaultLocale)
+      #end
+      #if ("$docLocale" != '' &amp;&amp; $docLocale != $xcontext.locale)
+        #set ($docUrl = $recentDoc.getURL('view', "language=${docLocale}"))
```

Comment:
I'm not sure if we should use something else than the `Locale.toString` method.

Reply:
toString() will display the locale id which is what we are doing in the edit panel, on the other hand in the drawer we are displaying the locale display name (see https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/menus_language.vm#L34)

Reply:
Ok I can keep this then.